### PR TITLE
Fix UG and DG issues

### LINKED ISSUE #197: [PE-D][Tester D] UG example commands fail
The UG guide included an example command as follows:

However, testing of this input led to an error

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: emmannyyy/ped#2

### LINKED ISSUE #198: [PE-D][Tester A] Incorrect example of addpa command
The example provided for adding a public address in the Try it Out! section of the UserGuide could be changed to follow the actual implementation of the addpa command!

Provided example to try: `addpa c/ETH n/Travis w/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2`

Command Format: `addpa INDEX c/NETWORK l/LABEL pa/PUBLIC_ADDRESS`

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: Haris-Irfan/ped#1

### LINKED ISSUE #199: [PE-D][Tester D] Default address given exceeds max length
User guide suggests this example command:

However, this has an address length more than 44, resulting in an error

Users who are not familiar and are trying out the app may not have an example address to test using the app

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: emmannyyy/ped#3

### LINKED ISSUE #206: [PE-D][Tester A] Incorrect example provided in addpa command guide
According to the screenshot, the following input would allow me to add the specified public address to the first contact in my address book. However, when I keyed this input into the command line, I encountered the following error.

I am not sure if this is intentional, as they did not specify the length of the public address field in the command details, but if that is the case, it would be good if this example was modified!

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: Haris-Irfan/ped#2

### LINKED ISSUE #209: [PE-D][Tester C] The example given in the user guide is not accepted
```
addpa 1 c/ETH l/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2
```
is an example given in the user guide but the length of the address exceeds 44 characters therefore i get error message
```
Public Address for length BTC/ETH/SOL should be less than 44 characters
```

-------------
Labels: `type.DocumentationBug` `severity.High`
original: Wxy2003-xy/ped#5

### LINKED ISSUE #223: [PE-D][Tester E] The example command in UG under "Try it Out!" is incorrect
This is how it looks in "Try it Out!"

This is the correct implementation:

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: foothaddeus/ped#4

### LINKED ISSUE #229: [PE-D][Tester D] Suggested example command does not work
Parameters for n/Name does not work.
Parameter w/Wallet also does not work. 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: emmannyyy/ped#12

### LINKED ISSUE #231: [PE-D][Tester E] Command word for searchpa is wrong in UG under feature description
Under feature description:

Correct implementation:

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: foothaddeus/ped#7

### LINKED ISSUE #233: [PE-D][Tester E] searchpa does not show network and wallet it belongs to if multiple personal addresses were found
-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: foothaddeus/ped#8

### LINKED ISSUE #234: [PE-D][Tester C] Command in the user guide is not recognized
publicAddressSearch stated in the user guide does not work, but searchpa, mentioned later in the summary.

-------------
Labels: `type.DocumentationBug` `severity.High`
original: Wxy2003-xy/ped#9

### LINKED ISSUE #235: [PE-D][Tester B] Duplicate public addess and wallet_name/label allowed for different network
Similar the the duplicate public address allowed for differnt wallet_name/label one, I was able to add the same wallet and public address under a different network

If this was the intended feature and allowed, do ignore this bug report!

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: The0nlyJuan/ped#9

### LINKED ISSUE #237: [PE-D][Tester D] publicAddressSearch command does not work, command not recognised
UG specifications

Actual behaviour

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: emmannyyy/ped#14

### LINKED ISSUE #238: [PE-D][Tester D] Attempted delete when person has no wallets provides invalid error message
-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: emmannyyy/ped#15

### LINKED ISSUE #244: [PE-D][Tester C] Inconsistency of examples in the user guide
The accepted command is addpa but the example given is addPublicAddress

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: Wxy2003-xy/ped#11

### LINKED ISSUE #253: [PE-D][Tester E] Example given for addpa command is invalid
This is one of the example commands given under addpa:

This does not work as intended, as the pa length is too long and an error message is returned:

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: foothaddeus/ped#13

### LINKED ISSUE #258: [PE-D][Tester D] Suggested example for DG does not work
Actual behaviour:

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: emmannyyy/ped#21

### LINKED ISSUE #268: [PE-D][Tester A] publicAddressSearch command not working
No matter what public address I use from my contacts list I always keep encountering an unknown command error.

This is your current implementation above.

This is what I am getting when I search for the address as shown above.

Another example of an available address that exists in my contacts list.

Hope y'all can fix this!

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Haris-Irfan/ped#16

### LINKED ISSUE #269: [PE-D][Tester A] Incorrect example provided for publicAddressSearch command
The address given in the example is significantly larger than 44 characters and as a result, would not work given your addpa command implementation. Would be good to change it so it is more accurate!

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: Haris-Irfan/ped#17

### PR DESCRIPTION
**Pull request by travisim**
Fixes #197, Fixes #198, Fixes #199, Fixes #206, Fixes #209, Fixes #223, Fixes #229, Fixes #231, Fixes #233, Fixes #234, Fixes #235, Fixes #237, Fixes #238, Fixes #244, Fixes #253, Fixes #258, Fixes #268, Fixes #269

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -560,6 +595,16 @@ Searches for a public address.
 
 Format: `searchpa pa/PUBLIC_ADDRESS`
 
+
+<box type="tip" seamless>
+
+The field PUBLIC_ADDRESS is not cap sensitive.
+This command only searches the current list of public addresses displayed(eg if all contacts is displayed
```
Comment (chongtzezhao) [inline docs/UserGuide.md:602]: nit: (grammar) change "if all contacts is displayed" to "if all contacts are displayed"

**Review by chongtzezhao — APPROVED** [review]
LGTM. Good job Travis!

File: docs/UserGuide.md (hunk 1 above)
Comment (travisim) [reply docs/UserGuide.md:602]: yes sir this shall be fixed

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -437,6 +438,10 @@ otherwise)
 4. AddressBook deletes the person
 
 ~~~~
+add to future enhancements
+The command 'forgiving' these extraneous inputs (i.e., giving an output same as or similar to if those inputs are not
+present) is not incorrect. You can mention in the UG that such inputs will be ignored. AB3 already does a similar thing
+for some commands. Any special handling of such inputs can be left as a future enhancement.
```
Comment (Airiinnn) [inline docs/DeveloperGuide.md:444]: Is this meant to be here?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -192,7 +216,7 @@ DLTbook comes with sample data to help you get started. Here are some basic comm
 3. **Add a DLT public address**
 
    ```
-   addpa 1 c/BTC l/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2
+   addpa c/ETH n/Travis w/wallet1 pa/0x0b1c9e1fb5e13c797c7f0134641810e9a7ca14d2
```
Comment (Airiinnn) [inline docs/UserGuide.md:219]: Isn't this the wrong format?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -560,6 +595,17 @@ Searches for a public address.
 
 Format: `searchpa pa/PUBLIC_ADDRESS`
 
+
+<box type="tip" seamless>
+
+The field PUBLIC_ADDRESS is not cap sensitive.
```
Comment (Airiinnn) [inline docs/UserGuide.md:601]: Standardise to case sensitive? Or doesn't matter

File: docs/UserGuide.md [hunk 5]
```diff
@@ -674,9 +733,9 @@ the data of your previous AddressBook home folder.
...
 | **Help**                    | `help`                                                                                                                                                                 |
 | **Exit**                    | `exit`                                                                                                                                                                 |
-| **Add Public Address**      | `addpa INDEX c/NETWORK l/WALLET_NAME pa/PUBLIC_ADDRESS`<br> e.g., `addpa 1 c/ETH l/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2`      |
-| **Edit Public Address**     | `editpa INDEX c/NETWORK l/WALLET_NAME pa/NEW_ADDRESS`<br> e.g., `editpa 3 c/BTC l/Daily wallet pa/14qViLJfdGaP4EeHnDyJbEGQysnCpwk3gd`                                  |
+| **Add Public Address**      | `addpa c/NETWORK n/NAME w/WALLET_NAME pa/PUBLIC_ADDRESS`<br> e.g., `addpa c/ETH n/Travis w/wallet1 pa/0x0b1c9e1fb5e13c797c7f0134641810e9a7ca14d2`                      |
```
Comment (Airiinnn) [inline docs/UserGuide.md:736]: Wrong format again?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -674,9 +733,9 @@ the data of your previous AddressBook home folder.
...
+| **Edit Public Address**     | `editpa INDEX c/NETWORK l/WALLET_NAME pa/NEW_ADDRESS`<br> e.g., `editpa 3 c/BTC l/Daily wallet pa/bc1q5y5960gr9vnjlmwfst232z07surun7rey5svu9`                          |
 | **Retrieve Public Address** | `retrievepa l/WALLET_NAME [c/NETWORK] [n/PERSON_NAME]`<br> e.g., `retrievepa l/wallet1 c/BTC n/John`                                                                   |
-| **Delete Public Address**   | `deletepa c/NETWORK [l/WALLET_NAME]`<br> e.g., `deletepa 1 c/BTC l/wallet1`                                                                                            |
-| **Public Address Search**   | `searchpa pa/PUBLIC_ADDRESS`<br> e.g., `searchpa pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2`                                                |
+| **Delete Public Address**   | `deletepa c/NETWORK [w/WALLET_NAME]`<br> e.g., `deletepa 1 c/BTC w/wallet1`                                                                                            |
```
Comment (Airiinnn) [inline docs/UserGuide.md:739]: Wrong field

**Review by Airiinnn — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (travisim) [reply docs/DeveloperGuide.md:444]: no its supposed to be in UG

File: docs/UserGuide.md (hunk 4 above)
Comment (travisim) [reply docs/UserGuide.md:601]: apologies, I shall use case-insensitive

File: docs/UserGuide.md (hunk 6 above)
Comment (travisim) [reply docs/UserGuide.md:739]: fixed this

File: docs/UserGuide.md (hunk 5 above)
Comment (travisim) [reply docs/UserGuide.md]: yes, fixed

File: docs/UserGuide.md (hunk 3 above)
Comment (travisim) [reply docs/UserGuide.md]: apologies again

File: docs/UserGuide.md [hunk 7]
```diff
@@ -674,9 +730,9 @@ the data of your previous AddressBook home folder.
...
 | **Help**                    | `help`                                                                                                                                                                 |
 | **Exit**                    | `exit`                                                                                                                                                                 |
-| **Add Public Address**      | `addpa INDEX c/NETWORK l/WALLET_NAME pa/PUBLIC_ADDRESS`<br> e.g., `addpa 1 c/ETH l/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2`      |
-| **Edit Public Address**     | `editpa INDEX c/NETWORK l/WALLET_NAME pa/NEW_ADDRESS`<br> e.g., `editpa 3 c/BTC l/Daily wallet pa/14qViLJfdGaP4EeHnDyJbEGQysnCpwk3gd`                                  |
+| **Add Public Address**      | `addpa INDEX c/NETWORK l/WALLET_NAME pa/PUBLIC_ADDRESS`<br> e.g., `addpa 1 c/ETH l/wallet1 pa/0x0b1c9e1fb5e13c797c7f0134641810e9a7ca14d2`                              |z
```
Comment (Nicholascyx) [inline docs/UserGuide.md:733]: There is a "z" at the end here.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -390,82 +416,84 @@ Examples:
...
   contain alphanumeric characters.
   This field is **case-insensitive** and will be converted to lowercase.
 
 Examples:
 
-* `addpa 1 c/ETH l/wallet1 pa/0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2` adds a public address
-  to the first person on the list `Alex Yeoh` under the ETH network with the wallet name `wallet1` and the public
-  address `0x28f91d6e72eaf4372892e6c6e45dc41b574163e9fcdf94f4997958b46d772fa2`.
+* `addpa 1 c/ETH l/wallet1 pa/0x0b1c9e1fb5e13c797c7f0134641810e9a7ca14d2
+* ` adds a public address
```
Comment (Nicholascyx) [inline docs/UserGuide.md:466]: Maybe adds a public address should be connected with the command, instead of being a new point?

**Review by Nicholascyx — COMMENTED** [review]
LGTM, all public addresses are now correct examples to be used.

File: docs/UserGuide.md (hunk 7 above)
Comment (travisim) [reply docs/UserGuide.md:730]: that is fixed

File: docs/UserGuide.md (hunk 8 above)
Comment (travisim) [reply docs/UserGuide.md:466]: good catch

**Pull request merged**